lightblue/src/main/java/org/esbtools/eventhandler/lightblue/LightblueNotificationRepository.java

```diff
@@ -0,0 +1,119 @@
+/*
+ *  Copyright 2016 esbtools Contributors and/or its affiliates.
+ *
+ *  This file is part of esbtools.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.esbtools.eventhandler.lightblue;
+
+import com.redhat.lightblue.client.LightblueClient;
+import com.redhat.lightblue.client.Locking;
+import com.redhat.lightblue.client.request.DataBulkRequest;
+import com.redhat.lightblue.client.response.LightblueException;
+
+import org.esbtools.eventhandler.EventHandlerException;
+import org.esbtools.eventhandler.FailedNotification;
+import org.esbtools.eventhandler.Notification;
+import org.esbtools.eventhandler.NotificationRepository;
+import org.esbtools.lightbluenotificationhook.NotificationEntity;
+
+import java.time.Clock;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class LightblueNotificationRepository implements NotificationRepository {
+    private final LightblueClient lightblue;
+    private final String[] entities;
+    private final Locking locking;
+    private final NotificationFactory notificationFactory;
+    private final Clock clock;
+
+    public LightblueNotificationRepository(LightblueClient lightblue, String[] entities,
+            String lockingDomain, NotificationFactory notificationFactory, Clock clock) {
+        this.lightblue = lightblue;
+        this.entities = entities;
+        this.notificationFactory = notificationFactory;
+        this.clock = clock;
+
+        locking = lightblue.getLocking(lockingDomain);
+    }
+
+    @Override
+    public List<Notification> retrieveOldestNotificationsUpTo(int maxEvents)
+            throws LightblueException {
+        try {
+            // TODO: Either block until lock acquired or throw exception and let caller poll
```

Comment:
Yep! Although it was somewhat of a question, actually. Should this method block, polling until the lock is acquired (with some timeout?) or should the caller deal with polling this method?

It's not a huge deal either way. If this method throws then it just needs to be explicitly documented that that is a reason it can throw. The tradeoff is more about implementations having to worry about polling until they can acquire lock or if they can fail fast and let caller deal with it.